packages/flutter/lib/src/cupertino/adaptive_text_selection_toolbar.dart

```diff
@@ -0,0 +1,375 @@
+  /// {@endtemplate}
+  /// {@template flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.editableText}
+  /// * [CupertinoAdaptiveTextSelectionToolbar.editableText], which builds the
+  ///   default Cupertino children for an [EditableText].
+  /// {@endtemplate}
+  /// {@template flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.selectable}
+  /// * [CupertinoAdaptiveTextSelectionToolbar.selectable], which builds the
+  ///   Cupertino children for content that is selectable but not editable.
+  /// {@endtemplate}
+  /// {@template flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.selectableRegion}
+  /// * [CupertinoAdaptiveTextSelectionToolbar.selectableRegion], which builds
+  ///   the default Cupertino children for a [SelectableRegion].
+  /// {@endtemplate}
+  const CupertinoAdaptiveTextSelectionToolbar({
+    super.key,
+    required this.children,
+    required this.primaryAnchor,
+    this.secondaryAnchor,
+    final TargetPlatform? targetPlatform,
+  }) : buttonItems = null,
+       _targetPlatform = targetPlatform;
+
+  /// Create an instance of [CupertinoAdaptiveTextSelectionToolbar] whose
+  /// children will be built from the given [buttonItems].
+  ///
+  /// See also:
+  ///
+  /// {@template flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.new}
+  /// * [CupertinoAdaptiveTextSelectionToolbar.new], which takes the children
+  ///   directly as a list of widgets.
+  /// {@endtemplate}
+  /// {@macro flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.editable}
+  /// {@macro flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.editableText}
+  /// {@macro flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.selectable}
+  /// {@macro flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.selectableRegion}
+  const CupertinoAdaptiveTextSelectionToolbar.buttonItems({
+    super.key,
+    required this.buttonItems,
+    required this.primaryAnchor,
+    this.secondaryAnchor,
+    final TargetPlatform? targetPlatform,
+  }) : children = null,
+       _targetPlatform = targetPlatform;
+
+  /// Create an instance of [CupertinoAdaptiveTextSelectionToolbar] with the
+  /// default children for an editable field.
+  ///
+  /// See also:
+  ///
+  /// * [AdaptiveTextSelectionToolbar.editable], which is similar to this but
+  ///   includes Material and Cupertino toolbars.
+  /// {@macro flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.new}
+  /// {@macro flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.buttonItems}
+  /// {@macro flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.editableText}
+  /// {@macro flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.selectable}
+  /// {@macro flutter.cupertino.CupertinoAdaptiveTextSelectionToolbar.selectableRegion}
+  CupertinoAdaptiveTextSelectionToolbar.editable({
+    super.key,
+    required bool readOnly,
+    required ClipboardStatus clipboardStatus,
+    required bool copyEnabled,
```

Comment:
Would it make sense to use nullable callbacks as indicator whether each functionality is enabled and remove the four `Enabled*` parameters?
```
    required VoidCallback? onCopy,
    required VoidCallback? onCut,
    required VoidCallback? onPaste,
    required VoidCallback? onSelectAll,
```
Then either check for `onCut != null` or maybe define getters like:
```
  bool get _isCutEnabled => onCut != null
```

Reply:
Yeah I like that better, thanks 👍